app/knight.py

```diff
@@ -0,0 +1,63 @@
+from __future__ import annotations
+
+from app.armour import Armour
+from app.weapon import Weapon
+from app.potion import Potion
+
+
+class Knight:
+    def __init__(
+            self,
+            name: str,
+            power: int,
+            hp: int,
+            armours: list,
+            weapon: dict,
+            potion: dict
+    ) -> None:
+        self.name = name
+        self.power = power
+        self.hp = hp
+        self.armour = self.set_armours(armours)
+        self.weapon = Weapon(weapon["name"], weapon["power"])
```

Comment:
tip: you can use unpacking here using **. Today at 14:00 QnA I'm gonna show a few examples ;)